src/error.rs

```diff
@@ -126,6 +126,12 @@ pub enum Error {
     ///  - it was negative
     ///  - it was too long
     InvalidCrlNumber,
+
+    /// A serial number was invalid:
+    ///  - it was misencoded
+    ///  - it was negative
+    ///  - it was too long
+    InvalidSerialNumber,
```

Comment:
BTW, for both `InvalidSerialNumber` and `InvalidCrlNumber` it would be nice to stick it in some kind of alphabetic order as much as possible (or maybe include a commit to reorder up front/in a separate PR?).

Reply:
I reordered up-front in a separate commit before adding any new entries. It felt reasonable to do that as part of this PR.